.github/workflows/check-c10-sync.yml

```diff
@@ -0,0 +1,32 @@
+name: check-c10-sync
+
+on:
+  pull_request:
+    paths:
+      - .ci/docker/ci_commit_pins/pytorch.txt
+      - .ci/scripts/compare_dirs.sh
+      - .ci/scripts/diff_c10_mirror_with_pytorch.sh
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}-${{ github.event_name == 'workflow_dispatch' }}-${{ github.event_name == 'schedule' }}
+  cancel-in-progress: true
+
+jobs:
+  check-c10-sync:
+    permissions:
+      id-token: write
+      contents: read
+    name: check-c10-sync
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Clone PyTorch
+        run: |
+          set -ex
+          git clone https://github.com/pytorch/pytorch.git
+          pushd pytorch || return
+          git checkout "$(< ../.ci/docker/ci_commit_pins/pytorch.txt)"
+          popd
```

Comment:
Given `check_c10_sync.sh` already handles this, why do we need to do this here?

Reply:
whoops